### PR TITLE
Skipping doctest on slow connection

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Currently a lot of our Azure builds are failing for the bert doctest due to slow connection of the servers.
Implements some custom logic that will skip the doctest if the download is timing out.

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/functional/text/bert.py [hunk 1]
```diff
@@ -30,16 +30,23 @@
...
-else:
-    __doctest_skip__ = ["bert_score"]
 
+    def _try_download() -> None:
+        AutoTokenizer.from_pretrained(_DEFAULT_MODEL)
+        AutoModel.from_pretrained(_DEFAULT_MODEL)
 
-# Default model recommended in the original implementation.
-_DEFAULT_MODEL = "roberta-large"
+    if _check_download_timeout(_try_download, default_timeout=150):
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:46]:
```suggestion
    if _check_download_timeout(_try_download, default_timeout=120):
```

File: src/torchmetrics/text/bert.py [hunk 2]
```diff
@@ -21,10 +21,18 @@
...
 
 if _TRANSFORMERS_AVAILABLE:
-    from transformers import AutoTokenizer
+    from transformers import AutoModel, AutoTokenizer
+
+    def _try_download() -> None:
+        AutoTokenizer.from_pretrained(_DEFAULT_MODEL)
+        AutoModel.from_pretrained(_DEFAULT_MODEL)
+
+    if _check_download_timeout(_try_download, default_timeout=150):
```
Comment (Borda) [inline src/torchmetrics/text/bert.py:34]:
```suggestion
    if _check_download_timeout(_try_download, default_timeout=120):
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
Really nice way! :chipmunk:

**Review by justusschock — APPROVED** [review]

**Review by stancld — APPROVED** [review]

File: src/torchmetrics/utilities/checks.py [hunk 3]
```diff
@@ -746,3 +749,45 @@ def is_overridden(method_name: str, instance: object, parent: object) -> bool:
         raise ValueError("The parent should define the method")
 
     return instance_attr.__code__ != parent_attr.__code__
+
+
+def _in_doctest() -> bool:
```
Comment (Borda) [inline src/torchmetrics/utilities/checks.py:754]: do we need this?

**Comment by Borda** [comment]
Paradox is that the GPU tests are hanging on doctests... 🐿️

**Comment by SkafteNicki** [comment]
> Paradox is that the GPU tests are hanging on doctests... 🐿️

Yes, but they are not hanging on Bert doctest anymore. Trying to add the skipping to other download intensive metrics.

File: src/torchmetrics/functional/text/bert.py [hunk 4]
```diff
@@ -30,16 +30,24 @@
...
-    __doctest_skip__ = ["bert_score"]
 
+    def _try_download() -> None:
+        """Download intensive operations."""
+        AutoTokenizer.from_pretrained(_DEFAULT_MODEL)
+        AutoModel.from_pretrained(_DEFAULT_MODEL)
 
-# Default model recommended in the original implementation.
-_DEFAULT_MODEL = "roberta-large"
+    if _check_download_timeout(_try_download, default_timeout=120):
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:47]: This shall also only for testing so we do download it even for TM import...

File: src/torchmetrics/utilities/checks.py [hunk 5]
```diff
@@ -749,3 +755,48 @@ def is_overridden(method_name: str, instance: object, parent: object) -> bool:
...
+    """Determine if script running in doctest context.
+
+    From:
+    https://stackoverflow.com/questions/27884404/printing-test-execution-times-and-pinning-down-slow-tests-with-py-test
```
Comment (Borda) [inline src/torchmetrics/utilities/checks.py:764]:
```suggestion
```
does not seems to be right reference

**Pull request merged**